python/ql/src/experimental/dataflow/internal/DataFlowPublic.qll

```diff
@@ -90,10 +103,14 @@ class CfgNode extends Node, TCfgNode {
  * to multiple `ExprNode`s, just like it may correspond to multiple
  * `ControlFlow::Node`s.
  */
-class ExprNode extends Node { }
+class ExprNode extends CfgNode {
```

Comment:
At this point in time, it would seem that `ExprNode` and `CfgNode` coincide. I'm guessing we're doing it this way so that we can potentially add in some of the non-`Expr` control flow nodes at a later date?

Reply:
That is correct, they are currently the same. That seems sufficient with the current implementation, but it is possible that we want to add more control flow nodes when some of the enhancing PRs land.

The reason we have `ExprNode` is that it is required by the shared dataflow library. See [here](https://github.com/github/codeql/blob/main/docs/ql-libraries/dataflow/dataflow.md#node-subclasses). Btw, I should probably check that document for updates again and see if other clean-up tasks present themselves :-)